### ISSUE TITLE
Classes, Annotations, Methods, Interfaces and other constructs need direct link to online Java doc or other web docs.

### ISSUE THREAD

**Issue by dmitrilc** [vaadin/docs#1428]
Java Classes, Annotations, Methods, Interfaces, and other constructs need to have direct links to the corresponding online Java doc version.

It can be confusing for developers when they encounter some vanilla-Java-sounding class names, but they turn out to be a Vaadin class. 

**StreamResource** is an example.
https://vaadin.com/docs/latest/flow/advanced/dynamic-content/#using-streamresource
https://vaadin.com/api/platform/23.0.10/com/vaadin/flow/server/StreamResource.html

If adding a link everywhere seems redundant, at least add a link for type definitions at the first occurrence in the current tutorial/guide.

The same can be said for non-Java classes and methods as well, such as TypeScript/JavaScript/CSS/HTML, etc.

**Comment by jouni** [comment]
Yeah, this would be super. I’m just wondering how to best implement this. Making it into a Design System Publisher feature sounds either restrictive (we’d need to make too many assumptions about customer API doc deployments) or elaborate (we’d need to consider multiple variables in customer API deployments).

I wonder if it would suffice if we inject links after the page load, using JavaScript? That could be done in the DSP theme, and we can make all the assumptions we want just for our own vaadin.com deployment.

**Comment by tarekoraby** [comment]
> I wonder if it would suffice if we inject links after the page load, using JavaScript?

How about adding normal Asciidoc links?

**Comment by jouni** [comment]
Well, maybe. I thought that might be very tedious, but perhaps it’s on a similar level of work, as we in any case have to explicitly define what keyword matches to what API docs link.

To keep the links working across version changes, the URLs should utilize the `{moduleMavenVersion:com.vaadin:vaadin}` replacement, to link to the same API docs version as the version used in the docs branch. For example:

```asciidoc
https://vaadin.com/api/platform/{moduleMavenVersion:com.vaadin:vaadin}/com/vaadin/flow/component/button/Button.html[[classname]`Button`]
```

Verbose, but that is what is needed.

Though, I wonder if there’s a way to create some kind of “macro” for this in AsciiDoc, that abstracts most of it.
